test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -115,16 +117,17 @@ describe('mainTemplate.json file - ', () => {
             });
         });
 
-        it('apiVersions must NOT be retrieved using providers().apiVersions[n].  This function is non-deterministic', () => {
+        // providers().apiVersions[n] MUST not be present
+        describe('apiVersions must NOT be retrieved using providers().apiVersions[n].  This function is non-deterministic', () => {
             templateFileJSONObjects.forEach(templateJSONObject => {
```

Comment:
@bmoore-msft can you confirm if this test needs to be run for ALL template files?

Reply:
Synced with @bmoore-msft, this is for ALL template files.